gateway-ha/src/main/java/io/trino/gateway/ha/module/RouterBaseModule.java

```diff
@@ -26,17 +25,19 @@
 import io.trino.gateway.ha.router.ResourceGroupsManager;
 import org.jdbi.v3.core.Jdbi;
 
+import static java.util.Objects.requireNonNull;
+
 public class RouterBaseModule
-        extends AppModule<HaGatewayConfiguration, Environment>
+        extends AbstractModule
 {
     final ResourceGroupsManager resourceGroupsManager;
     final GatewayBackendManager gatewayBackendManager;
     final QueryHistoryManager queryHistoryManager;
     final JdbcConnectionManager connectionManager;
 
-    public RouterBaseModule(HaGatewayConfiguration configuration, Environment environment)
+    public RouterBaseModule(HaGatewayConfiguration configuration)
     {
-        super(configuration, environment);
+        requireNonNull(configuration, "config is null");
```

Comment:
The  message should be "configuration is null", but please take a look at another review comment.